src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/TypeExtensions.cs

```diff
@@ -13,14 +21,41 @@ public static bool IsTypeOf(this TypeDesc type, string ns, string name)
             return type is MetadataType mdType && mdType.Name == name && mdType.Namespace == ns;
         }
 
-        public static bool IsDeclaredOnType(this MethodDesc method, string ns, string name)
+        public static bool IsTypeOf(this TypeDesc type, string fullTypeName)
```

Comment:
I'll look into this some more - I vaguely remember we changed this to match something in the analyzer, but now when I look at the code there it all does the parsing as well, so this doesn't feel right.

I'd like to do this in a followup change (change in the linker and then migrate that to native AOT), does that make sense?

Reply:
> I'd like to do this in a followup change

I wasn't expecting to (I wasn't even expecting the follow up change, because the perf regression is probably papercut level, but those things eventually lead to death by a thousand papercuts). I prefixed all I wrote with "Not really commenting on this PR because I know this code comes from a different repo - just a general sigh " :)